src/resource/solidDataset.ts

```diff
@@ -1133,3 +1137,37 @@ function resolveLocalIrisInThing(
       : thing.url,
   });
 }
+
+/**
+ * Fetch the contents of '.well-known/solid' for a given resource URL.
+ *
+ * @param url URL of a Resource.
+ * @param options Optional parameter `options.fetch`: An alternative `fetch` function to make the HTTP request, compatible with the browser-native [fetch API](https://developer.mozilla.org/docs/Web/API/WindowOrWorkerGlobalScope/fetch#parameters).
+ * @returns Promise resolving to a [[SolidDataset]] containing the data at '.well-known/solid' for the given Resource, or rejecting if fetching it failed.
+ */
+export async function getWellKnownSolid(
+  url: UrlString | Url,
+  options: Partial<
+    typeof internal_defaultFetchOptions & ParseOptions
+  > = internal_defaultFetchOptions
+): Promise<SolidDataset & WithServerResourceInfo> {
+  const urlString = internal_toIriString(url);
+  const resourceMetadata = await getResourceInfo(urlString, options);
+  const linkedResources = getLinkedResourceUrlAll(resourceMetadata);
+  const rootResources =
+    linkedResources["http://www.w3.org/ns/pim/space#storage"];
+  const rootResource = rootResources?.length === 1 ? rootResources[0] : null;
+  if (!rootResource) {
+    throw new SolidClientError(
+      `Unable to determine root resource for Resource at [${url}].`
+    );
+  }
+  const wellKnownSolidUrl = `${rootResource}.well-known/solid`;
```

Comment:
Even though this is valid because it only makes sense for the root resource to be a container, and its canonical IRI should always end with a trailing slash, would it be safer to use `new URL(".well-known/solid", rootResource)` instead of direct concatenation to make sure we get a valid IRI out of this in the case where the rootResource IRI is missing a trailing slash ?

Reply:
good point, done!